system/include/emscripten/bind.h

```diff
@@ -1906,6 +1913,36 @@ class_<std::vector<T>> register_vector(const char* name) {
         ;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// Optional
+////////////////////////////////////////////////////////////////////////////////
+
+#if defined(EMSCRIPTEN_HAS_OPTIONAL)
+namespace internal {
+
+template<typename T>
+struct OptionalAccess {
+    static bool has_value(const std::optional<T>& o) {
+        return o.has_value();
+    }
+
+    static T value(const std::optional<T> optional) {
+        return optional.value();
+    }
+};
+
+} // end namespace internal
+
+template<typename T>
+class_<std::optional<T>> register_optional(const char* name) {
+    return class_<std::optional<T>>(name)
```

Comment:
I think it would be more idiomatic to map it to `T | undefined`. E.g. this is what wasm-bindgen does for `Option<T>`.